benchmarks/benchmarks-core/src/jmh/java/io/micrometer/benchmark/core/MeterRegistrationContentionBenchmark.java

```diff
@@ -0,0 +1,104 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micrometer.benchmark.core;
+
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.simple.SimpleMeterRegistry;
+import java.util.concurrent.TimeUnit;
+import org.openjdk.jmh.annotations.Benchmark;
+import org.openjdk.jmh.annotations.BenchmarkMode;
+import org.openjdk.jmh.annotations.Fork;
+import org.openjdk.jmh.annotations.Group;
+import org.openjdk.jmh.annotations.GroupThreads;
+import org.openjdk.jmh.annotations.Level;
+import org.openjdk.jmh.annotations.Measurement;
+import org.openjdk.jmh.annotations.Mode;
+import org.openjdk.jmh.annotations.OutputTimeUnit;
+import org.openjdk.jmh.annotations.Scope;
+import org.openjdk.jmh.annotations.State;
+import org.openjdk.jmh.annotations.TearDown;
+import org.openjdk.jmh.annotations.Warmup;
+import org.openjdk.jmh.runner.Runner;
+import org.openjdk.jmh.runner.RunnerException;
+import org.openjdk.jmh.runner.options.Options;
+import org.openjdk.jmh.runner.options.OptionsBuilder;
+
+@Measurement(iterations = 5, time = 1)
+@Warmup(iterations = 10, time = 1)
+@Fork(3)
+@BenchmarkMode(Mode.SampleTime)
+@OutputTimeUnit(TimeUnit.MICROSECONDS)
+@State(Scope.Group)
+public class MeterRegistrationContentionBenchmark {
+    MeterRegistry registry = new SimpleMeterRegistry();
+
+    /** Ensures we are testing a clean registry each time */
+    @TearDown(Level.Invocation)
+    public void clear() {
+        registry.clear();
+    }
+
+    @Benchmark @Group("no_contention") @GroupThreads(1)
+    public void no_contention_register_counter() {
+        register_counter();
+    }
+
+    @Benchmark @Group("mild_contention") @GroupThreads(2)
+    public void mild_contention_register_counter() {
+        register_counter();
+    }
+
+    @Benchmark @Group("high_contention") @GroupThreads(8)
+    public void high_contention_register_counter() {
+        register_counter();
+    }
+
+    /** @see io.micrometer.benchmark.core.MeterRegistrationBenchmark#register_counter() */
+    void register_counter() {
+        registry.counter("counter", "k", "v");
+    }
+
+    @Benchmark @Group("no_contention") @GroupThreads(1)
+    public void no_contention_register_counter_redundant() {
+        register_counter_redundant();
+    }
+
+    @Benchmark @Group("mild_contention") @GroupThreads(2)
+    public void mild_contention_register_counter_redundant() {
+        register_counter_redundant();
+    }
+
+    @Benchmark @Group("high_contention") @GroupThreads(8)
+    public void high_contention_register_counter_redundant() {
+        register_counter_redundant();
+    }
+
+    /** @see io.micrometer.benchmark.core.MeterRegistrationBenchmark#register_counter_redundant() */
+    void register_counter_redundant() {
```

Comment:
Better might be to do `registry.counter(...)` in setup and only one of these in the benchmark. For the redundant case, we don't have to worry about any caveats with `Level.Invocation` either.

Reply:
you mean only do contention on redundant case? bc if we do counter in setup, then the one marked not redundant will be redundant right?

Reply:
Could strip down, but was thinking a separate class for redundant case so it has its own setup / teardown

Reply:
ic yeah that would work